### DISCUSSION TITLE
Can any llama models solve this???

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3238] [Q&A]
Any llama models... (what's the lowest (7b,13b) etc model that can solve this?
```
### System: You specialize in grammar and sentence spelling.. 

### User: Capitalize the first letter of each word in 'how jean-claude de la grande pierre uses iphone' correctly. 

### Assistant: How Jean-Claude De La Grande Pierre Uses iPhone
```

Chat GPT4 rendered correctly:

```
The title with appropriate capitalization would be:

"How Jean-Claude de la Grande Pierre Uses iPhone"
```

**Comment by verdooft** [comment]
I don't use original or smaller llama2 models, but i tried this:
```
./main -t 8 -m models/nous-hermes-llama2-70b.Q5_K_M.gguf -c 4096 --temp 0.7 --repeat_penalty 1.1 -n -1 -p " You specialize in grammar and sentence spelling. Capitalize the first letter of each word in 'how jean-claude de souza uses iphone' correctly.\n "

Answer: How Jean-Claude De Souza Uses iPhone
```

**Reply by verdooft** [reply]
Ok, i tested a 7b Model:
```
./main -t 8 -m ~/Downloads/models/llama2_7b_chat_uncensored.Q5_K_M.gguf -c 4096 --temp 0.7 --repeat_penalty 1.1 -n -1 -p "### HUMAN:\nYou specialize in grammar and sentence spelling. Capitalize the first letter of each word in 'how jean-claude de souza uses iphone' correctly.\n\n### RESPONSE:"

"How Jean-Claude De Souza Uses iPhone" [end of text]
```

**Reply by hiqsociety** [reply]
@verdooft "de la" should be small caps. look at my results from chatgpt4.

sorry, just re-updated my question. i copied the wrong prompts. supposed to have "de la".

**Comment by verdooft** [comment]
ok, but you gave "how jean-claude de souza uses iphone" as text and ChatGPT 4 wrote: How Jean-Claude de la Grande Pierre Uses iPhone

Different texts in input and output? There is no " la grande pierre" in the input.

**Reply by hiqsociety** [reply]
sorry, just re-updated my question. i copied the wrong prompts. supposed to have "de la".

**Reply by verdooft** [reply]
Ok, the 7b Model write uses with a small u:

```
./main -t 8 -m ~/Downloads/models/llama2_7b_chat_uncensored.Q5_K_M.gguf -c 4096 --temp 0.7 --repeat_penalty 1.1 -n -1 -p "### HUMAN:\nYou specialize in grammar and sentence spelling. Capitalize the first letter of each word in 'how jean-claude de la grande pierre uses iphone' correctly.\n\n### RESPONSE:"

How Jean-Claude de la Grande Pierre uses iPhone is grammatically correct without any spelling errors. [end of text]
```

uses is a verb, i think, writing it with a small u is correct. But the pitput depends on different setting like seed and temperature.

**Comment by bobqianic** [comment]
> "How Jean-Claude de la Grande Pierre Uses iPhone"

Why is this correct? I'm not a French speaker. Can anyone explain this?

**Reply by hiqsociety** [reply]
search for "Jean-Claude de la Grande Pierre" <- de la should be in small caps. on google etc.

**Reply by hiqsociety** [reply]
sorry, just re-updated my question. i copied the wrong prompts. supposed to have "de la".
